testeng/resources/cleanup-python-code-create-pr.sh

```diff
@@ -1,34 +1,49 @@
 #!/bin/bash
 set -e
 
-rm -rf code_cleanup_venv
-virtualenv --python=python$PYTHON_VERSION code_cleanup_venv -q
-source code_cleanup_venv/bin/activate
+declare -i RESULT=$?
 
-echo "Upgrading pip..."
-pip install pip==20.0.2
+IFS=',' read -ra SCRIPTSTORUN <<<"$SCRIPTS"
 
-echo "Getting current sha..."
-cd $REPO_NAME
-export CURRENT_SHA=$(git rev-parse HEAD)
+PACKAGESTOINSTALL=$(echo $PACKAGES | tr , " ")
 
-echo "Running cleanup..."
+IFS=',' read -ra REPOS <<<"$REPO_NAMES"
 
-PACKAGESTOINSTALL=`echo $PACKAGES | tr , " "`
+for repo in "${REPOS[@]}"; do
 
-pip install $PACKAGESTOINSTALL
+  rm -rf "${repo}"
 
-IFS=',' read -ra SCRIPTSTORUN <<< "$SCRIPTS"
-for element in "${SCRIPTSTORUN[@]}"; do
-  `echo "$element"`
-done
+  if git clone https://github.com/edx/"${repo}".git; then true; else RESULT=$? && continue; fi
+
+  rm -rf "${repo}"-code_cleanup_venv
+  virtualenv --python=python"$PYTHON_VERSION" "${repo}"-code_cleanup_venv -q
+  source "${repo}"-code_cleanup_venv/bin/activate
+
+  echo "Upgrading pip..."
+  if pip install pip==20.0.2; then true; else RESULT=$? && continue; fi
+
+  echo "Getting current sha..."
+  cd "${repo}"
+  export CURRENT_SHA=$(git rev-parse HEAD)
```

Comment:
I learned from the recent repo health dashboard trouble that this can fail if the repo is empty (such as a new one with no initial commit yet), since `HEAD` doesn't exist yet.